src/app/report/edit-display/edit-display.component.ts

```diff
@@ -209,20 +195,28 @@ export class EditDisplayComponent {
       checkpointId = this.report.uid.split('#')[1];
     }
 
-    const body = { stub: stubStrategy, ...this.getReportValues(checkpointId) };
+    const body = this.getReportValues(checkpointId);
 
     this.httpService.updateReport(storageId, body, this.currentView.storageName).subscribe({
       next: (response: any) => {
         response.report.xml = response.xml;
         this.report = response.report;
         this.saveReportEvent.next(this.report);
-        this.editor.setNewReport(this.report.xml);
+        this.editor.setNewReport(UpdateReportUtil.isUpdateCheckpoint(body) ? body.checkpointMessage : this.report.xml);
         this.disableEditing();
       },
       error: () => catchError(this.errorHandler.handleError()),
     });
   }
 
+  discardChanges(): void {
+    this.disableEditing();
+    if (this.report.uid) {
+      this.editor.setNewReport(this.report.message);
+    }
+    this.toastService.showSuccess('Changes discarded!');
+  }
```

Comment:
So what happens if you're editing a report and not a checkpoint, don't you also need to reset the edit-form then?

Reply:
No, because the user goes back to read-only mode, so the form will reset automatically